packages/protocol/test/L1/TaikoL1.integration.test.ts

```diff
@@ -0,0 +1,160 @@
+            const { tx, commit } = await commitBlock(taikoL1, block);
+
+            // blockMetadata is inputs[0], txListBytes = inputs[1]
+            const config = await taikoL1.getConfig();
+            const gasLimit = config[7];
+            await proposeBlock(
+                taikoL1,
+                block,
+                commit.txListHash,
+                tx.blockNumber as number,
+                0,
+                block.gasLimit
+            );
+
+            await expect(
+                proposeBlock(
+                    taikoL1,
+                    block,
+                    commit.txListHash,
+                    tx.blockNumber as number,
+                    0,
+                    gasLimit.add(1)
+                )
+            ).to.be.revertedWith("L1:gasLimit");
+        });
+
+        it("should revert with invalid extraData", async function () {
+            const block = await l2Provider.getBlock("latest");
+            const { tx, commit } = await commitBlock(taikoL1, block);
+
+            const meta: BlockMetadata = {
+                id: 0,
+                l1Height: 0,
+                l1Hash: ethers.constants.HashZero,
+                beneficiary: block.miner,
+                txListHash: commit.txListHash,
+                mixHash: ethers.constants.HashZero,
+                extraData: ethers.utils.hexlify(ethers.utils.randomBytes(33)), // invalid extradata
+                gasLimit: block.gasLimit,
+                timestamp: 0,
+                commitSlot: 1,
+                commitHeight: tx.blockNumber as number,
+            };
+
+            const inputs = buildProposeBlockInputs(block, meta);
+
+            await expect(taikoL1.proposeBlock(inputs)).to.be.revertedWith(
+                "L1:extraData"
+            );
+        });
+
+        it("should commit and be able to propose", async function () {
+            const block = await l2Provider.getBlock("latest");
+            const { tx, commit } = await commitBlock(taikoL1, block);
+
+            await proposeBlock(
+                taikoL1,
+                block,
+                commit.txListHash,
+                tx.blockNumber as number,
+                0,
+                block.gasLimit
+            );
+
+            const stateVariables = await taikoL1.getStateVariables();
+            const nextBlockId = stateVariables[4];
+            const proposedBlock = await taikoL1.getProposedBlock(
+                nextBlockId.sub(1)
+            );
+
+            expect(proposedBlock[0]).not.to.be.eq(ethers.constants.HashZero);
+            expect(proposedBlock[2]).not.to.be.eq(ethers.constants.AddressZero);
+            expect(proposedBlock[3]).not.to.be.eq(BigNumber.from(0));
+
+            const isCommitValid = await taikoL1.isCommitValid(
+                1,
+                tx.blockNumber as number,
+                commit.hash
+            );
+
+            expect(isCommitValid).to.be.eq(true);
```

Comment:
we should also test:

- no more than the max slot number of blocks can be proposed
- if a proposed block is verified, it shall not be returned by `getProposedBlock` -- I think there is a bug in the contract code as it doesn't check the boundary of the block id param -- https://github.com/taikoxyz/taiko-mono/pull/11679